docker/setup_demo.sh

```diff
@@ -20,8 +20,8 @@ SCRIPT_PATH=$(cd `dirname $0`; pwd)
 HUDI_DEMO_ENV=$1
 WS_ROOT=`dirname $SCRIPT_PATH`
 COMPOSE_FILE_NAME="docker-compose_hadoop284_hive233_spark353_amd64.yml"
-if [ "$HUDI_DEMO_ENV" = "--mac-aarch64" ]; then
-  COMPOSE_FILE_NAME="docker-compose_hadoop284_hive233_spark353_arm64.yml"
+if [ "$(uname -m)" = "arm64" ]; then
```

Comment:
@deepakpanda93 in follow up pr, clean up this `HUDI_DEMO_ENV` as you made it unused. Also update any docs (search in branch asf-site) that shows how to use this script accordingly.